lib/semian/net_http.rb

```diff
@@ -87,14 +88,24 @@ def disabled?
       raw_semian_options.nil?
     end
 
+    def fatal_server_errors?
```

Comment:
This should not be a public method. Since you only use it in one place, I'd inline it.